File: tests/fixtures/pure-ts/src/events.ts
type Handler<T> = (payload: T) => void
export class EventBus {
  private handlers = new Map<string, Handler<unknown>[]>()
  on(event: string, handler: Handler<unknown>): void {
    const list = this.handlers.get(event) ?? []
    list.push(handler)
    this.handlers.set(event, list)
  }
}

File: tests/fixtures/pure-py-typed/src/events.py
from typing import Callable, Any

class EventBus:
    def __init__(self) -> None:
        self._handlers: dict[str, list[Callable[..., Any]]] = {}

    def on(self, event: str, handler: Callable[..., Any]) -> None:
        self._handlers.setdefault(event, []).append(handler)
